src/main/kotlin/blackjack/model/Dealer.kt

```diff
@@ -0,0 +1,24 @@
+package blackjack.model
+
+class Dealer(
+    scoreCalculator: ScoreCalculator,
+) : Participant(scoreCalculator) {
+    fun drawIfNeeded(cardDeck: CardDeck): Int {
+        var count = INITIAL_RESULT_COUNT
+
+        while (score() <= DEALER_DRAW_CRITERIA && !isBust()) {
+            draw(cardDeck)
+            count++
+        }
+
+        return count
+    }
+
+    fun firstVisibleCard(): List<Card> = cards.take(DEALER_FIRST_SHOWN_COUNT)
```

Comment:
```suggestion
    fun firstVisibleCard(): Card = cards.first()
```

이렇게 하는 게 조금 더 가독성에 좋지 않을까 하고 보니까 `List <Card>` 로 한 이유가 `View에서 표시하기 좋게 하기 위함` 으로 보이는데요 Model 관점에서 봤을 때 View에서 어떻게 표시가 되기 편한지를 고려를 해야 할지 아니면 현재 함수에 맞게끔 하는 게 좋을까요?

---

그리고 `firstVisibleCard` 보다는 현재 함수가 어떠한 행동을 해야 하는지를 접두사로 잘 표현해 주면 좋을 것 같습니다.

Reply:
> 이렇게 하는 게 조금 더 가독성에 좋지 않을까 하고 보니까 `List <Card>` 로 한 이유가 `View에서 표시하기 좋게 하기 위함` 으로 보이는데요 Model 관점에서 봤을 때 View에서 어떻게 표시가 되기 편한지를 고려를 해야 할지 아니면 현재 함수에 맞게끔 하는 게 좋을까요?
> 
> 그리고 `firstVisibleCard` 보다는 현재 함수가 어떠한 행동을 해야 하는지를 접두사로 잘 표현해 주면 좋을 것 같습니다.

showInitialCards를 abstract로 두면서 List<Card>는 유지되어야 할 것 같아요!
이를 위해 플레이어와 딜러 모두 다른 함수를 가지는 것이 더 부자연스럽다고 생각되어서요.

처음에는 저도 딜러에만 처음 카드를 보여줘도 괜찮지 않나? 라는 생각을 했는데


> 처음 카드를 받고 딜러는 1장을 공개하고 플레이어는 2장을 공개한다.
> 이것이 지금 블랙잭의 비즈니스 로직 중 하나인데요. 해당 로직은 공개한다. 라는 행위는 동일하지만 내부 구현은 서로 다를 것 같아요. 이런 것들을 한 번 abstract를 사용해서 관리 해보는 것도 좋은 연습이 될 것 같습니다.

이 부분을 보니까 충분히 '비즈니스 로직으로 2장을 보여준다' 가 납득이 되어서
abstract fun 을 선택하고 List<Card>를 반환하는 함수로 추상화를 진행하겠습니다!